### DISCUSSION TITLE
namespace name 'IImageInfo' could not be found

### DISCUSSION THREAD

**Discussion by daryl-williams** [SixLabors/ImageSharp#2506] [Q&A] [answered]
I am trying to evaluate the ImageSharp library for possible use in an application where I work, however as I worked my way though the "Getting Started" page when I got to the "Identify Image" section with the following example:

```
IImageInfo imageInfo = Image.Identify(@"image.jpg");
Console.WriteLine($"Width: {imageInfo.Width}");
Console.WriteLine($"Height: {imageInfo.Height}");

```

the build failed with the following error:

`error CS0246: The type or namespace name 'IImageInfo' could not be found (are you missing a using directive or an assembly reference?)
`

Searching for how to fix this problem led me to this comment in stackoverflow:

`Please install the Nuget Package "SixLabor.ImageSharp" in the desired project. Once done, in your Startup.cs configure it as mentioned on their getting started page https://docs.sixlabors.com/articles/imagesharp.web/gettingstarted.html`

which led me to the Six Labors/ImageSharp "Setup and Configuration" page which in turn let me to google for more information on DotNet "Setup and Configuration" which led me to a search page that was full of references to ASP.NET.

At this point i should mention that I come from a UNIX background and am quite new to the DotNet environment. I am using the the dotnet command line and working with net6.0, Imagesharp version 3.0.1 and the program I need to write has nothing to do with ASP or the web and will just be a simple command line filter program.

Any help resolving this issue would be much appreciated.

Regards,

Daryl

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah sorry, that's a mistake. The type is [ImageInfo](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ImageInfo.html). I've fixed the sample docs.

**Comment by daryl-williams** [comment]
Hi @JimBobSquarePants and thanks for your reply,  I appreciate the information.

D

**Comment by daryl-williams** [comment]
Hi @james_m_south, sorry to bother and I'm not sure if you'll see this but I wanted to let you know that it looks like some documentation might also need updating. I found references to IImageInfo at:

https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.html

and the different format pages it references.

D

**Reply by JimBobSquarePants** [reply]
There’s no reference to `IImageInfo` on that page. The API docs are generated from the code.

**Comment by daryl-williams** [comment]
Sorry I was reading up on ImageInfo and followed some links to learn about PixelTypeInfo and must have confused myself along the way seeing references to IImageFormat, IImageDecode, and so forth. I'm assuming now the II* is a convention referring an interface. Sorry for the bother.

**Reply by JimBobSquarePants** [reply]
No worries at all 🙂
